clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockOutputStream.java

```diff
@@ -259,10 +259,10 @@ private void ensureNotFinished() {
     @Override
     public void close() throws IOException {
         if (!finished) {
+            // basicallly flush the buffer writing the last block
             writeBlock();
+            // write the end block and finish the stream
             writeEndMark();
-            flush();
```

Comment:
OK. So strictly speaking this patch changes behavior in the sense that `out.flush()` is not guaranteed to be called. I would have kept it just to avoid unnecessary problem in the future. I really don't think an extra line of `out.flush()` would be a problem. But since the patch is already committed, I would't bother to change it as well since there is very little chance this can be a problem.